### PR TITLE
Add tracepoints within createProgram

### PR DESCRIPTION
**Pull request by amcasey**
These were useful during the Midgard investigation.

### CODE REVIEW

**Comment by amcasey** [comment]
Draft since I'll need to rebase this when @elibarzilay's PR goes in.

File: src/compiler/program.ts [hunk 1]
```diff
@@ -2747,6 +2768,16 @@ namespace ts {
             resolvedTypeReferenceDirective?: ResolvedTypeReferenceDirective,
             refFile?: RefFile
         ): void {
+            tracing.begin(tracing.Phase.Program, "processTypeReferenceDirective", { directive: typeReferenceDirective, hasResolved: !!resolveModuleNamesReusingOldState, refKind: refFile?.kind, refPath: refFile?.file.path });
```
Comment (sheetalkamat) [inline src/compiler/program.ts:2771]: Shouldnt this be inside resolveTypeReferenceDirectiveNamesWorker instead of  here like resolveModuleNamesWorker ?

Comment (amcasey) [reply src/compiler/program.ts:2771]: I added this tracepoint so I could figure out whether one type references was substantially more expensive than the others.  I can also add one in `resolveTypeReferenceDirectiveNamesWorker` though.

**Review by orta — APPROVED** [review]
Very cool

**Review by elibarzilay — APPROVED** [review]

**Pull request merged**